lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3SelectInputFile.java

```diff
@@ -0,0 +1,166 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.s3;
+
+import io.trino.filesystem.Location;
+import io.trino.filesystem.TrinoInput;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.TrinoInputStream;
+import software.amazon.awssdk.core.exception.SdkException;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.ExpressionType;
+import software.amazon.awssdk.services.s3.model.HeadObjectRequest;
+import software.amazon.awssdk.services.s3.model.HeadObjectResponse;
+import software.amazon.awssdk.services.s3.model.InputSerialization;
+import software.amazon.awssdk.services.s3.model.NoSuchKeyException;
+import software.amazon.awssdk.services.s3.model.OutputSerialization;
+import software.amazon.awssdk.services.s3.model.RequestPayer;
+import software.amazon.awssdk.services.s3.model.SelectObjectContentRequest;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.Instant;
+
+import static java.util.Objects.requireNonNull;
+
+public class S3SelectInputFile
+        implements TrinoInputFile
+{
+    private final S3Client s3;
+    private final S3AsyncClient asyncS3;
+    private final String query;
+    private final S3Location location;
+    private final boolean enableScanRange;
+    private final InputSerialization inputSerialization;
+    private final OutputSerialization outputSerialization;
+    private Long length;
+    private Instant lastModified;
+    private final RequestPayer requestPayer;
+
+
+    public S3SelectInputFile(
+            S3Client s3,
+            S3AsyncClient asyncS3,
+            S3Context context,
+            S3Location location,
+            String query,
+            boolean enableScanRange,
+            InputSerialization inputSerialization,
+            OutputSerialization outputSerialization)
+    {
+        this.s3 = requireNonNull(s3, "s3 is null");
+        this.asyncS3 = requireNonNull(asyncS3, "asyncS3 is null");
+        this.query = requireNonNull(query, "query is null");
+        this.location = requireNonNull(location, "location is null");
+        location.location().verifyValidFileLocation();
+        this.requestPayer = context.requestPayer();
+        this.enableScanRange = enableScanRange;
+        this.inputSerialization = requireNonNull(inputSerialization, "inputSerialization is null");
+        this.outputSerialization = requireNonNull(outputSerialization, "outputSerialization is null");
+    }
+
+    @Override
+    public TrinoInput newInput()
+    {
+        return new S3SelectInput(location(), asyncS3, buildSelectObjectContentRequest());
+    }
+
+    @Override
+    public TrinoInputStream newStream()
+    {
+        return null; // new S3InputStream(location(), client, newGetObjectRequest(), length);
```

Comment:
This seems backwards. Since S3 Select is only for text formats, we should only need the input stream version. `TrinoInput` is only used by ORC and Parquet.